lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -2450,15 +2449,15 @@ ParseResult FIRStmtParser::parseNode() {
   //
   // TODO: This optimization doesn't respect annotated, temporary nodes.
   Value result;
-  if (actualName) {
+  if (!actualName.empty()) {
     ArrayAttr annotations = builder.getArrayAttr({});
-    getAnnotations(getModuleTarget() + ">" + actualName.getValue(),
-                   annotations);
-    result = builder.create<NodeOp>(info.getLoc(), initializer, actualName,
-                                    annotations);
+    auto name = builder.getStringAttr(actualName);
+    getAnnotations(getModuleTarget() + ">" + actualName, annotations);
+    result =
+        builder.create<NodeOp>(info.getLoc(), initializer, name, annotations);
```

Comment:
note to self: try passing in a stringref instead of stringattr

Reply:
Random FYI if you didn't already know, but keeping things as StringAttr is a bit more efficient than passing around StringRef's.  The StringRef ultimately has to be uniqued back into an attr before it can be stored on an op.  If you're computing a new name then you need to go through this, but if you're just passing names around it can be helpful to keep them as attrs.

Reply:
Makes sense! I looked through the code trying to see if I unintentionally made things slower, and I think it's ok.  As you were getting at, most of the parser code is creating the attributes from scratch already, so passing in a StringRef I is OK.  My first attempt at this change I was creating StringAttrs manually absolutely everywhere, and I slowly reduced the clutter by passing in StringRefs instead.